[Bug]: Can not build on Mac OS PostgresApp

### What happened?

clang diagnosed error: /Applications/Postgres.app/Contents/Versions/17/include/postgresql/server/c.h:59:10: fatal error: 'stdio.h' file not found

### pgvectorscale extension affected

0.5.1

### PostgreSQL version used

17.0

### What operating system did you use?

OSX 15.2 

### What installation method did you use?

Source

### What platform did you run on?

Not applicable

### Relevant log output and stack trace

```bash
clang diagnosed error: /Applications/Postgres.app/Contents/Versions/17/include/postgresql/server/c.h:59:10: fatal error: 'stdio.h' file not found
```

### How can we reproduce the bug?

```bash
cargo pgrx install --release
```

### Are you going to work on the bugfix?

None

--- a/DEVELOPMENT.md
+++ b/DEVELOPMENT.md
@@ -1,6 +1,6 @@
 # Setup your pgvectorscale developer environment
 
-You build pgvectorscale from source, then integrate the extension into each database in your PostgreSQL environment. 
+You build pgvectorscale from source, then integrate the extension into each database in your PostgreSQL environment.
 
 ## pgvectorscale prerequisites
 
@@ -12,22 +12,10 @@ To create a pgvectorscale developer environment, you need the following on your 
     ```
     sudo apt-get install make gcc pkg-config clang postgresql-server-dev-16 libssl-dev
     ```
-  
+
 * [Rust][rust-language]:
     ```shell
     curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
-    ```
-  
-* [Cargo-pgrx][cargo-pgrx]:
-    ```shell
-    cargo install --locked cargo-pgrx --version $(cargo metadata --format-version 1 | jq -r '.packages[] | select(.name == "pgrx") | .version')
-    ```
-  You must reinstall cargo-pgrx whenever you update Rust, cargo-pgrx must 
-  be built with the same compiler as pgvectorscale.
-
- * The pgrx development environment:
-    ```shell
-    cargo pgrx init --pg16 pg_config
     ```
 
 ## Build and install pgvectorscale on your database
@@ -39,11 +27,32 @@ To create a pgvectorscale developer environment, you need the following on your 
     cd pgvectorscale/pgvectorscale
     ```
 
+1. Install [Cargo-pgrx][cargo-pgrx]:
+
+   ```bash
+   cargo install --locked cargo-pgrx --version $(cargo metadata --format-version 1 | jq -r '.packages[] | select(.name == "pgrx") | .version')
+   ```
+   You must reinstall cargo-pgrx whenever you update Rust, cargo-pgrx must be
+   built with the same compiler as pgvectorscale.
+
+1. Initialize the pgrx development environment:
+
+  ```bash
+  cargo pgrx init --pg16 pg_config
+  ```
+
 1. Build pgvectorscale:
 
-    ```shell
-    cargo pgrx install --release
-    ```
+   ```shell
+   cargo pgrx install --release
+   ```
+
+   If the destination folder requires elevated permissions, use the `--sudo`
+   flag:
+
+   ```shell
+   cargo pgrx install --sudo --release
+   ```
 
 1. Connect to the database:
 
@@ -57,6 +66,41 @@ To create a pgvectorscale developer environment, you need the following on your 
     CREATE EXTENSION IF NOT EXISTS vectorscale CASCADE;
     ```
 
+# Troubleshooting
+
+## MacOS: fatal error: 'stdio.h' file not found
+
+Make sure you have the command line tools:
+
+```bash
+xcode-select --install
+```
+
+If the error persists try setting explicitly the `SDKROOT` environment
+variable:
+
+```bash
+export SDKROOT=$(xcrun --sdk macosx --show-sdk-path)
+```
+
+Make sure the variable was set:
+
+
+```bash
+$ env | grep SDKROOT
+SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
+```
+
+## Permission denied while installing the extension artifacts
+
+If the PostgreSQL directory where the extension needs to be installed requires
+elevated permissions (for example, when using MacOS PostgreSQL that installs in the
+`/Applications` folder, or when installing from a package manager that uses a system
+directory), you need to use the pgrx `--sudo` flag:
+
+```bash
+cargo pgrx install --sudo --release
+```
 
 [pgvector]: https://github.com/pgvector/pgvector/blob/master/README.md
 [rust-language]: https://www.rust-lang.org/

